aptos-move/aptos-release-builder/src/simulate.rs

```diff
@@ -0,0 +1,500 @@
+//! the governance process to be skipped altogether.
+//!
+//! In other words, this simulation is intended for checking whether a governance
+//! proposal will execute successfully, assuming it gets approved.
+
+use crate::aptos_framework_path;
+use anyhow::{anyhow, bail, Result};
+use aptos::{
+    common::types::PromptOptions, governance::compile_in_temp_dir, move_tool::FrameworkPackageArgs,
+};
+use aptos_gas_schedule::{AptosGasParameters, FromOnChainGasSchedule, ToOnChainGasSchedule};
+use aptos_language_e2e_tests::account::AccountData;
+use aptos_move_debugger::aptos_debugger::AptosDebugger;
+use aptos_rest_client::Client;
+use aptos_types::{
+    account_address::AccountAddress,
+    account_config::ChainIdResource,
+    on_chain_config::{GasScheduleV2, OnChainConfig},
+    state_store::{
+        in_memory_state_view::InMemoryStateView, state_key::StateKey,
+        state_storage_usage::StateStorageUsage, state_value::StateValue,
+        Result as StateStoreResult, StateView, TStateView,
+    },
+    transaction::{ExecutionStatus, Script, TransactionArgument, TransactionStatus},
+    write_set::{TransactionWrite, WriteSet},
+};
+use aptos_vm::{data_cache::AsMoveResolver, AptosVM};
+use aptos_vm_logging::log_schema::AdapterLogSchema;
+use clap::Parser;
+use move_binary_format::{
+    access::ModuleAccess,
+    file_format::{
+        AddressIdentifierIndex, Bytecode, FunctionDefinition, FunctionHandle, FunctionHandleIndex,
+        IdentifierIndex, ModuleHandle, ModuleHandleIndex, Signature, SignatureIndex,
+        SignatureToken, Visibility,
+    },
+    CompiledModule,
+};
+use move_core_types::{
+    identifier::{IdentStr, Identifier},
+    language_storage::{ModuleId, StructTag},
+    move_resource::MoveResource,
+    resolver::ModuleResolver,
+};
+use once_cell::sync::Lazy;
+use std::{
+    collections::HashMap,
+    io::Write,
+    path::{Path, PathBuf},
+    sync::Mutex,
+};
+use url::Url;
+
+/***************************************************************************************************
+ * Compiled Module Helpers
+ *
+ **************************************************************************************************/
+fn find_function_def_by_name<'a>(
+    m: &'a mut CompiledModule,
+    name: &IdentStr,
+) -> Option<&'a mut FunctionDefinition> {
+    for (idx, func_def) in m.function_defs.iter().enumerate() {
+        let func_handle = m.function_handle_at(func_def.function);
+        let func_name = m.identifier_at(func_handle.name);
+        if name == func_name {
+            return Some(&mut m.function_defs[idx]);
+        }
+    }
+    None
+}
+
+fn get_or_add<T: PartialEq>(pool: &mut Vec<T>, val: T) -> usize {
+    match pool.iter().position(|elem| elem == &val) {
+        Some(idx) => idx,
+        None => {
+            let idx = pool.len();
+            pool.push(val);
+            idx
+        },
+    }
+}
+
+fn get_or_add_addr(m: &mut CompiledModule, addr: AccountAddress) -> AddressIdentifierIndex {
```

Comment:
nit: why not spell address and identifier fully?

Reply:
Probably just personal preference -- found the names getting a little bit too long